jax/_src/api.py

```diff
@@ -228,15 +229,16 @@ def _infer_argnums_and_argnames(
 
 
 def jit(
-  fun: Callable,
-  *,
-  static_argnums: Union[int, Iterable[int], None] = None,
-  static_argnames: Union[str, Iterable[str], None] = None,
-  device: Optional[xc.Device] = None,
-  backend: Optional[str] = None,
-  donate_argnums: Union[int, Iterable[int]] = (),
-  inline: bool = False,
-) -> stages.Wrapped:
+    fun: Callable,
+    *,
+    static_argnums: Union[int, Iterable[int], None] = None,
+    static_argnames: Union[str, Iterable[str], None] = None,
+    device: Optional[xc.Device] = None,
+    backend: Optional[str] = None,
+    donate_argnums: Union[int, Iterable[int]] = (),
+    inline: bool = False,
+    abstracted_axes: Optional[Any] = None,
```

Comment:
What is your plan for landing some of this API change before the dynamic shapes support is ready? We can call this `experimental_abstracted_axes` and document it as not ready for use. Or, we can add a special private _jit that exposes this API.

Reply:
I'm comfortable just with leaving it undocumented. Having a private `_jit` would be fine too though; we could probably reuse `_python_jit` for that purpose.